File: android/src/main/kotlin/com/smile/iana_time_zone/IanaTimeZonePlugin.kt
package com.smile.iana_time_zone

import androidx.annotation.NonNull

import android.content.Context
import android.os.Build
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result
import java.util.TimeZone


/** IanaTimeZonePlugin */
class IanaTimeZonePlugin: FlutterPlugin, MethodCallHandler {
  /// The MethodChannel that will the communication between Flutter and native Android
  ///
  /// This local reference serves to register the plugin with the Flutter Engine and unregister it
  /// when the Flutter Engine is detached from the Activity
  private lateinit var channel : MethodChannel

  override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
    channel = MethodChannel(flutterPluginBinding.binaryMessenger, "iana_time_zone")
    channel.setMethodCallHandler(this)
  }

  override fun onMethodCall(call: MethodCall, result: Result) {
//    if (call.method == "getPlatformVersion") {
//      result.success("Android ${android.os.Build.VERSION.RELEASE}")
//    } else {
//      result.notImplemented()
//    }
    if (call.method == "getIanaTimeZone") {
      result.success(getIanaTimeZone())
    } else {
      result.notImplemented()
    }
  }

  private fun getIanaTimeZone(): String {
    val timeZone = TimeZone.getDefault()
    return timeZone.id // This returns IANA time zone identifier
  }

  override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
    channel.setMethodCallHandler(null)
  }
}

File: ios/Classes/IanaTimeZonePlugin.swift
import Flutter
import UIKit

public class IanaTimeZonePlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "iana_time_zone", binaryMessenger: registrar.messenger())
    let instance = IanaTimeZonePlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    if (call.method == "getIanaTimeZone") {
      result(getIanaTimeZone())
    } else {
      result(FlutterMethodNotImplemented)
    }
  }

  private func getIanaTimeZone() -> String {
    let timeZone = TimeZone.current
    return timeZone.identifier // This returns IANA time zone identifier
  }
}